extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/hll/StripedBufferLock.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.aggregation.datasketches.hll;
+
+import com.google.common.util.concurrent.Striped;
+
+import java.util.concurrent.locks.ReadWriteLock;
+
+/**
+ * Utility for locking positions in a buffer.
+ */
+public class StripedBufferLock
```

Comment:
I wonder does maybe have wider utility than just this extension? Nothing seems specific to datasketches, should we move this to core to make this available to other aggregators? Though, if we drop the locks from buffer/vector aggs, maybe it isn't needed.

Reply:
I deleted it from the latest commit based on the conversation in https://github.com/apache/druid/pull/11115#discussion_r614568145.